adam/ontology/phase1_ontology.py

```diff
@@ -213,7 +215,11 @@ def subtype(sub: OntologyNode, _super: OntologyNode) -> None:
 # Information about the hierarchical structure of objects
 # is given at the end of this module because it is so bulky.
 
-INANIMATE_OBJECT = OntologyNode("inanimate-object", inheritable_properties=[INANIMATE])
+INANIMATE_OBJECT = OntologyNode(
+    "inanimate-object",
+    inheritable_properties=[INANIMATE],
+    non_inheritable_properties=[PERSON_CAN_HAVE],
```

Comment:
Why here, since it is not inherited and we never instantiate a raw `INANIMATE_OBJECT` anyway?

Reply:
The curriculum for PERSON_HAS_OBJECT instantiates a raw `INANIMATE_OBJECT`. You can find this in `phase1_curriculum.py` line 166. I can change this to a `BALL` and therefore could remove this addition.

Reply:
@LichMaster98 : The interpretation of

```
object_variable("object_0", INANIMATE_OBJECT)
```

is that the variable ahs to be bound to a sub-type of `INANIMATE_OBJECT` which has the `CAN_FILL_TEMPLATE_SLOT` (IIRC) property, so `INANIMATE_OBJECT` itself will never be instantiated.

We will want to change this to
```
object_variable("object_0", INANIMATE_OBJECT,
     required_properties=[PERSON_CAN_HAVE])
```

(from memory, may not be exactly right)